x-pack/plugins/security_solution/public/management/pages/endpoint_hosts/store/middleware.ts

```diff
@@ -611,18 +613,19 @@ async function endpointDetailsMiddleware({
   if (listData(getState()).length === 0) {
     const { page_index: pageIndex, page_size: pageSize } = uiQueryParams(getState());
     try {
-      const response = await coreStart.http.post<HostResultList>(HOST_METADATA_LIST_ROUTE, {
-        body: JSON.stringify({
-          paging_properties: [{ page_index: pageIndex }, { page_size: pageSize }],
-        }),
+      const response = await coreStart.http.get<MetadataListResponse>(HOST_METADATA_LIST_ROUTE, {
+        query: {
+          page: pageIndex,
+          pageSize,
+        },
       });
-      response.request_page_index = Number(pageIndex);
+
       dispatch({
         type: 'serverReturnedEndpointList',
         payload: response,
       });
 
-      dispatchIngestPolicies({ http: coreStart.http, hosts: response.hosts, store });
+      dispatchIngestPolicies({ http: coreStart.http, hosts: response.data, store });
```

Comment:
I was initially trying to migrate this to react query and hooks but there are some other dependencies on the data here in the middleware so probably best to scope that into its own ticket.